cmd/metalbond-speaker/main.go

```diff
@@ -88,6 +90,19 @@ func main() {
 		os.Exit(1)
 	}
 
+	connectionDeadline := time.Now().Add(connectionTimeoutDuration)
+	for time.Now().Before(connectionDeadline) {
```

Comment:
Can you use `wait.PollUntilContect()` instead of using a for loop.

Reply:
Sure, adapted.